core/commands/dag/dag.go

```diff
@@ -552,6 +553,10 @@ The output of blocks happens in strict DAG-traversal, first-seen, order.
 			return err
 		}
 
+		ctx, cancel := context.WithCancel(req.Context)
+		ctx = exchange.NewSession(req.Context)
```

Comment:
Not sure if I'm following this right, should this be
```suggestion
		ctx = exchange.NewSession(ctx)
```